packages/firestore/scripts/assertion-id-tool.ts

```
@@ -0,0 +1,372 @@
+      type: "boolean",
+    })
+    .option("check", {
+      alias: "C",
+      describe: "Check for duplicate usage of assertion ids",
+      type: "boolean",
+    })
+    .check((argv) => {
+      // Enforce mutual exclusivity among options *within* the scan command
+      const options = [argv.F, argv.L, argv.N, argv.C].filter(Boolean).length;
+      if (options > 1) {
+        throw new Error("Options -F, -L, -N, -C are mutually exclusive.");
+      }
+      return true;
+    })
+    .help()
+    .alias('help', 'h')
+    .strict()
+    .parse(); // Execute parsing
+
+  // Extract directory path (safe due to demandOption)
+  const targetDirectory = path.resolve(argv['dir'] as string);
+
+  // set verbosity
+  isVerbose = !!argv['verbose'];
+
+  // Validate directory
+  try {
+    const stats = fs.statSync(targetDirectory);
+    if (!stats.isDirectory()) {
+      console.error(`Error: Provided path is not a directory: ${targetDirectory}`);
+      process.exit(1);
+    }
+  } catch (error: any) {
+    console.error(`Error accessing directory ${targetDirectory}: ${error.message}`);
+    process.exit(1);
+  }
+
+  log(`Scanning directory: ${targetDirectory}`);
+  const filesToScan = getTsFilesRecursive(targetDirectory);
+
+  if (filesToScan.length === 0) {
+    log("No relevant .ts or .tsx files found.");
+    process.exit(0);
+  }
+  log(`Found ${filesToScan.length} files. Analyzing for assertion ids...`);
+
+  const allOccurrences = findFunctionCalls(filesToScan);
+  log(`Scan complete. Found ${allOccurrences.length} potential assertion id occurrences.`);
+
+  // Determine action based on flags
+  if (argv['find']) {
+    handleFind(allOccurrences, argv['find']);
+  } else if (argv['new']) {
+    handleNew(allOccurrences);
+  } else if (argv['check']) {
+    handleCheck(allOccurrences);
+  } else {
+    // Default action is List (-L or no flag)
+    handleList(allOccurrences);
+  }
+}
+
+// Run the main function
+main().catch(error => {
```

Can you just let the exception bubble up and crash the program?

---

yeah, implemented